Update README to clarify recalculation processes (#85)
ebi-gene-expression-group/bulk-recalculations: README.md

./run_sorting_hat_test_data.sh EXPS_DIR
```

The experiments path contains one or more directories with Atlas accession names E-* (e.g. `E-MTAB-5577`), having at least configuration files in xml format after curation process.

Optionally, worflow execution can be tailored to specific accessions or species by defining these variables in the sorting-hat script.

### New experiment processing and re-processing

Completed processing by iRAP Single Lib (ISL) is necessary before new experiment processing. It will run all rules avilable for the experiment type.

### Recalculations

> This is necessary for the Ensembl Update (E! Update) part of a Data Release. E! Update brings annotations from Biomart, E! Mysql databases and ftp sites for all the relevant organisms in Expressiona Atlas, and leaves them in a format that can be consumed for the decoration process and the web applications. Ensembl Update validators (for Biomart attributes and GTF URL validations) are performed before running recalculations.

Reviewer: Ensembl Update validators also looks for existence of MySQL database

Author: [ecd4644](https://github.com/ebi-gene-expression-group/bulk-recalculations/pull/85/commits/ecd4644ebe53e6bb354d40505410bcc211cb11f4)

> For differial experiments, the following outputs are generated (which correspond to rules):

Reviewer: ```suggestion
For differential RNA-seq experiments, the following outputs are generated (which correspond to rules):
```

Author: [ecd4644](https://github.com/ebi-gene-expression-group/bulk-recalculations/pull/85/commits/ecd4644ebe53e6bb354d40505410bcc211cb11f4)

1. **Percentile ranks rule**
- Output: `{accession}-percentile-ranks.tsv`

2. **Differential tracks rule**
- Outputs: `{accession}.{contrast_id}.genes.pval.bedGraph`
- Outputs: `{accession}.{contrast_id}.genes.log2foldchange.bedGraph`

3. **Differential GSEA rule**
- Outputs: `{accession}.{contrast_id}.{ext_db}.gsea.tsv`
- Outputs: `{accession}.{contrast_id}.{ext_db}.gsea_list.tsv`

4. **atlas_experiment_summary rule**
- Output: `{accession}-atlasExperimentSummary.Rdata`

For baseline RNA-seq experiments, the following outputs are generated:

1. **Baseline tracks rule**
- Outputs: `{accession}.{assay_id}.genes.expressions_{metric}.bedGraph`
  (where metric could be fpkm/tpm)

2. **Baseline heatmap rule**
- Outputs: `{accession}-heatmap-{metric}.pdf`
- Output: `{accession}-heatmap.pdf`

3. **Baseline coexpression rule**
- Outputs: `{accession}-{metric}-coexpressions.tsv.gz`
- Output: `{accession}-coexpressions.tsv.gz`

4. **atlas_experiment_summary rule**
- Output: `{accession}-atlasExperimentSummary.Rdata`

For proteomics experiments (`proteomics_baseline`, `proteomics_baseline_dia`, `proteomics_differential`), recalculations are not implemented.

> 

Reviewer: ```suggestion
### Snakemake rules by experiment types
```

> | Rule                        |  Baseline  | Differential RNA-seq | Differential microarray |

Reviewer: ```suggestion
| Rule                        |  Baseline RNA-seq | Differential RNA-seq | Differential microarray |
```

Reviewer: or its for both RNA-seq and Microarray

Author: EA doesn't have baseline microarrays

|-----------------------------|------------|----------------------|-------------------------|
| atlas_experiment_summary    |  &#x2713;  |       &#x2713;       |         &#x2713;        |
| check_differential_gsea     |            |       &#x2713;       |         &#x2713;        |
| differential_gsea           |            |       &#x2713;       |         &#x2713;        |
| differential_tracks         |            |       &#x2713;       |         &#x2713;        |
| percentile_ranks            |            |       &#x2713;       |         &#x2713;        |
| baseline_coexpression       |  &#x2713;  |                      |                         |
| baseline_heatmap            |  &#x2713;  |                      |                         |
| baseline_tracks             |  &#x2713;  |                      |                         |
| link_baseline_coexpression  |  &#x2713;  |                      |                         |
| link_baseline_heatmap       |  &#x2713;  |                      |                         |

> | touch_inputs_baseline       |  &#x2713;  |                      |                         |

Reviewer: This is very helpful
